scripts/heratape_backup.py

```diff
@@ -0,0 +1,330 @@
+TESTING=True #if true, use test db, skip actual tape  writing.
+TAPESIZE = 18e12
+
+local_client = LibrarianClient(LOCAL_CONN_NAME)
+mydrive = 0 # tape drive selector. 0 or 1. TODO make this set the oddness of the jd 
+mydrive  = int(sys.argv[1])
+if not mydrive in [0,1]: 
+    logging.error('input tape drive must be 0 or 1')
+    sys.exit(1)
+
+# some AI bullshyt                                                                                                          
+logging.basicConfig(                                                                                                        
+    level=logging.DEBUG if os.environ.get('LOG_LEVEL') == 'DEBUG' else logging.INFO,                                        
+    format='%(asctime)s - %(levelname)s - %(message)s'                                                                      
+)                                                                                                                           
+logger = logging.getLogger('heratape-drive[{mydrive}]')                 
+                                                                                                                            
+                                                                                                                            
+                                                                                                                            
+logger.info("heratape_backup starting")   
+
+logger.info(f'using tape drive {mydrive}')
+logger.info('(tape 0 works on even JDs, 1 takes care off odd)')
+
+#accumulate 18TB worth of whole nights.
+#write them to a tape known to be blank
+# if interrupted, just try again
+# avoids writing multiple archives to a single tape, a practice which is avoided by seasoned veterans 
+backupsize = 0 # tracks total size of planned backup, in bytes
+while(True):
+    if backupsize ==0:
+        joined_results = [] #this is where we'll accumulate files to backup.
+        jds_to_backup = [] #keep a list of the JDS going into this backup
+        #otherwise we're in the process of adding days to the planned backup
+
+    # check for incomplete writes (identified as files with no write date)
+    incomplete_jds, incomplete_files = list_incomplete()
+    incomplete_jds = np.array(incomplete_jds)
+    if  np.any(incomplete_jds % 2 == mydrive):
+        logger.error(f'undated entries found in heratape.files for jds {incomplete_jds[incomplete_jds % 2 ==mydrive]}')
+        logger.error('This error might indicate that tape writing was interrupted. Check tape position.')
+        logger.error('TODO: add more helpful printout telling name of suspect tape')
+        logger.error('to proceed please delete these records and restart me')
+        sys.exit(1)
+
+    #   select JD from a priority scheme that is NOT in the backed up JD list
+    priority_list = '/users/djacobs/src/heratape/scripts/hera_backup_jd_priorities.txt'
```

Comment:
should point to the file you added to the repo in this PR, not your local path to it.